### PR TITLE
Runtime deprecation of `tagMap` & `typeMap` aliases

### PR DESCRIPTION
**Pull request by Avasam**
Closes https://github.com/pyasn1/pyasn1/issues/11

CC @tiran

### CODE REVIEW

File: pyasn1/codec/ber/decoder.py [hunk 1]
```diff
@@ -2186,3 +2182,12 @@ def _callSubstrateFunV4asV5(substrateFunV4, asn1Object, substrate, length):
...
 decode = Decoder()
+
+def __getattr__(attr: str):
+    if attr == "tagMap":
+        warnings.warn("tagMap is deprecated. Please use TAG_MAP instead.", DeprecationWarning)
+        return TAG_MAP
+    if attr == "typeMap":
+        warnings.warn("typeMap is deprecated. Please use TYPE_MAP instead.", DeprecationWarning)
+        return TYPE_MAP
+    raise AttributeError(attr)
```
Comment (Avasam) [inline pyasn1/codec/ber/decoder.py:2193]: I can probably shorten this to:
```suggestion
    deprecations = {"tagMap": "TAG_MAP", "typeMap": "TYPE_MAP"}
    if newAttr := deprecations.get(attr):
        warnings.warn(f"{attr} is deprecated. Please use {newAttr} instead.", DeprecationWarning)
        return globals()[newAttr]
    raise AttributeError(attr)
```
2 less lines, less repetitions.

File: pyasn1/type/base.py [hunk 2]
```diff
@@ -228,7 +228,7 @@ def plug(self, *args, **kw):
 
         return cls._instance
 
-    def __getattr__(self, attr):
+    def __getattr__(self, attr: str):
```
Comment (droideck) [inline pyasn1/type/base.py:231]: It's a minor nitpick, but I'd rather keep the commit changes exactly about the subject, nothing additional.
We can have more PRs in the future for type changes and this change together with other stuff. :)

**Review by droideck — APPROVED** [review]
Besides the minor nitpick, looks good! Thank you!

File: pyasn1/codec/ber/decoder.py (hunk 1 above)
Comment (droideck) [reply pyasn1/codec/ber/decoder.py:2193]: Sounds good to me!
